rollbar/lib/telemetry.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import threading
+import time
+
+import rollbar
+
+
+class Queue:
```

Comment:
I looked at the functionality of these, and I could not find what I was looking for. I need to return all the items with the predefined queue size. and if the new item makes the queue size bigger than the predefined queue size then the first item must be removed. Like I said, I did not see anything here that would work efficiently.

Reply:
From the [Python documentation](https://docs.python.org/3/library/collections.html#collections.deque):

> `class collections.deque([iterable[, maxlen]])`
> (...)
> If maxlen is not specified or is None, deques may grow to an arbitrary length. Otherwise, the deque is bounded to the specified maximum length. Once a bounded length deque is full, when new items are added, a corresponding number of items are discarded from the opposite end. Bounded length deques provide functionality similar to the tail filter in Unix. They are also useful for tracking transactions and other pools of data where only the most recent activity is of interest.

Reply:
yes, but it does not have a function to retrieve all the items. I guess I can do the list()

Reply:
There is no need to provide such a function to any of the Python containers. In fact, none of the built-in containers provide such a function. Python supports the iterator protocol (see [PEP 234](https://www.python.org/dev/peps/pep-0234/)).

> I guess I can do the list()

If you need a list, you can do so (thanks to the iterator protocol).